### PR TITLE
HPCC-15269 Fix missing child graph progress issue

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

**Comment by jakesmith** [comment]
May want to target earlier build...?

File: thorlcr/slave/slwatchdog.cpp [hunk 1]
```diff
@@ -98,7 +98,9 @@ class CGraphProgressHandlerBase : public CSimpleInterface, implements ISlaveWatc
                 ForEachItemIn(g, activeGraphs) // NB: 1 for each slavesPerProcess
                 {
                     CGraphBase &graph = activeGraphs.item(g);
-                    graph.serializeStats(progressData);
+                    progressData.append((unsigned)graph.queryJobChannel().queryMyRank()-1);
+                    if (!graph.serializeStats(progressData))
```
Comment (ghalliday) [inline thorlcr/slave/slwatchdog.cpp]: @jakesmith can you explain why this fixes it?  It isn't clear to me what difference this would have

Comment (jakesmith) [inline thorlcr/slave/slwatchdog.cpp]: It used to serialize the slave # per graph, inc. child graphs.
Now it sends it once per subgraph, serialized here, not in graph object, avoiding it being serialized per child graph as before, which the master did not expect when deserializing it.

**Comment by jakesmith** [comment]
@ghalliday back to you.

**Pull request merged**